src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -433,50 +452,70 @@ public synchronized void unsubscribe(UnsubscribeRequest request)
         return CompletableFuture.completedFuture(0);
     }
 
-    private synchronized void spoolMessage()  {
-        if (spoolingFuture.get() == null || spoolingFuture.get().isCancelled()) {
-            spoolingFuture.set(ses.scheduleWithFixedDelay(this::spoolTask, 0, 5, TimeUnit.SECONDS));
+    @SuppressWarnings("PMD.AvoidCatchingThrowable")
+    protected CompletableFuture<Integer> publishSingleSpoolerMessage() throws InterruptedException {
+        final long id = spool.popId();
+        PublishRequest request = spool.getMessageById(id);
+        if (request == null) {
+            return null;
+        }
+        MqttMessage m = new MqttMessage(request.getTopic(), request.getPayload());
```

Comment:
It's not difficult to imagine spooler implementations where `getMessageById` is interruptible. Can we move this inside the try block? If we don't, I expect we'll have a bug in the future where messages get dropped.